platforms/documentation/docs/src/snippets/bestPractices/avoidComputationInConfigurationPhase-do/groovy/build.gradle

```diff
@@ -0,0 +1,15 @@
+// tag::do-this[]
+abstract class MyTask extends DefaultTask {
```

Comment:
🤔 Moving from an adhoc task to a task type is a larger change.  I think we should minimize the difference between Avoid and Do by making both either typed or ad-hoc tasks.